modules/gitops-configure-rollout-route-traffic-using-openshift-service-mesh.adoc

```diff
@@ -0,0 +1,448 @@
+= Configuring Argo Rollouts to route traffic by using OpenShift Service Mesh
+
+You can use OpenShift Service Mesh to configure Argo Rollouts by creating a gateway, two kubernetes services - stable and canary which point to the pods within each version of the services, a virtualservice, and a rollout custom resource in the following example procedure.
+
+The rollout routes 20% of traffic to a canary version of the application, and waits for a manual promotion. Then it routes 40% of traffic, and waits for another manual promotion. It then performs multiple automated promotions until it routes the entire traffic to the new application version.
+
+.Procedure
+
+. Create a `Gateway` to accept the inbound traffic for your mesh.
+
+.. Create a YAML file, and copy the following YAML into it.
+The following example creates a gateway called `rollouts-demo-gateway`:
++
+[source,yaml]
+----
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: rollouts-demo-gateway #<1>
+spec:
+  selector:
+    istio: ingressgateway #<2>
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+----
+<1> The name of the gateway.
+<2> Specifies the name of the ingress gateway. The gateway configures exposed ports and protocols but does not include any traffic routing configuration.
+.. Apply the YAML file by running the following command.
++
+[source,yaml]
+----
+$ oc apply -f gateway.yaml
+----
+. Create the services for the canary and stable versions of the application.
+.. In the *Administrator* perspective of the web console, click *Networking* -> *Services*.
+
+.. Click *Create Service*.
+
+.. On the *Create Service* page, click *YAML view* and add the following snippet:
+The following example creates a stable service called `rollouts-demo-stable`. Stable traffic is directed to this service.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: rollouts-demo-stable
+spec:
+  ports: #<1>
+  - port: 80
+    targetPort: http
+    protocol: TCP
+    name: http
+  selector: #<2>
+    app: rollouts-demo
+----
+<1> Specifies the name of the port used by the application for running inside the container.
+<2> Ensure that the contents of the selector field are the same in stable service and Rollout custom resource (CR).
+
+.. Click *Create* to create a stable service.
+
+.. Repeat these steps to create the canary service:
+The following example creates a canary service called `rollouts-demo-canary`. Canary traffic is directed to this service.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  name: rollouts-demo-canary
+spec:
+  ports: #<1>
+  - port: 80
+    targetPort: http
+    protocol: TCP
+    name: http
+  selector: #<2>
+    app: rollouts-demo
+----
+<1> Specifies the name of the port used by the application for running inside the container.
+<2> Ensure that the contents of the `selector` field are the same in canary service and `Rollout` custom resource (CR).
```

Comment:
Same comment about how you can use just the CR abbreviation if you introduced it above:
```suggestion
<2> Ensure that the contents of the `selector` field are the same in canary service and `Rollout` CR.
```

Reply:
Fixed.